src/vs/editor/contrib/inlayHints/browser/inlayHintsHover.ts

```diff
@@ -154,9 +156,12 @@ export class InlayHintsHover extends MarkdownHoverParticipant implements IEditor
 			if (!this._languageFeaturesService.hoverProvider.has(model)) {
 				return AsyncIterableObject.EMPTY;
 			}
-			return getHover(this._languageFeaturesService.hoverProvider, model, new Position(range.startLineNumber, range.startColumn), token)
+			return getHover(this._languageFeaturesService.hoverProvider, model, new Position(range.startLineNumber, range.startColumn), undefined, token)
 				.filter(item => !isEmptyMarkdownString(item.hover.contents))
-				.map(item => new MarkdownHover(this, part.item.anchor.range, item.hover.contents, false, 2 + item.ordinal));
+				.map(item => {
+					item.hover.dispose();
```

Comment:
I looked at the code and saw that the inlayHintsHoverParticipant extends the MarkdownHoverParticipant, so it will call the `renderHoverParts` method in it, which already takes care of the disposal. Hence that line is not needed.